### PR TITLE
feat: print cmd for uploaded snapshots

### PR DESCRIPTION
**Pull request by kushsharma**
- added a new `print` command under `snapshot`
- print cmd will fetch snapshot from server and output to stdout as protobuf

also fixes  https://github.com/odpf/stencil/issues/67

### CODE REVIEW

File: storage/postgres/repository.go [hunk 1]
```diff
@@ -76,18 +76,22 @@ func (r *Store) UpdateSnapshotLatestVersion(ctx context.Context, snapshot *model
...
 		Namespace: namespace,
 		Name:      name,
 	}
 	var query strings.Builder
 	var args []interface{}
-	query.WriteString(`SELECT id, version, latest from snapshots where namespace=$1 and name=$2 and latest=$3`)
-	args = append(args, namespace, name, latest)
+	query.WriteString(`SELECT id, version, latest from snapshots where namespace=$1 and name=$2`)
+	args = append(args, namespace, name)
+	if latest != nil {
```
Comment (harikrishnakanchi) [inline storage/postgres/repository.go:88]: we no need to add latest=false condition.
```go
if latest {

}
```
check should suffice.

**Comment by harikrishnakanchi** [comment]
Hey, if possible can you separate printcmd changes from download request bug?

File: storage/postgres/repository.go [hunk 2]
```diff
@@ -76,18 +76,22 @@ func (r *Store) UpdateSnapshotLatestVersion(ctx context.Context, snapshot *model
...
+func (r *Store) GetSnapshotByFields(ctx context.Context, namespace, name, version string, latest *bool) (*models.Snapshot, error) {
 	sh := &models.Snapshot{
 		Namespace: namespace,
 		Name:      name,
 	}
 	var query strings.Builder
 	var args []interface{}
-	query.WriteString(`SELECT id, version, latest from snapshots where namespace=$1 and name=$2 and latest=$3`)
-	args = append(args, namespace, name, latest)
+	query.WriteString(`SELECT id, version, latest from snapshots where namespace=$1 and name=$2`)
```
Comment (chakravarthyvp) [inline storage/postgres/repository.go:86]: SQLs like this are prone to SQL injection.
Should we parameterize the SQL?
 If not, validate the params?

Comment (kushsharma) [reply storage/postgres/repository.go:86]: @chakravarthyvp we are parameterizing the SQL I think? Why do you think we are not doing it?

File: storage/postgres/repository.go (hunk 1 above)
Comment (kushsharma) [reply storage/postgres/repository.go:88]: @harikrishnakanchi you mean like this?
```go
if latest != nil && *latest == true {
    args = append(args, *latest)
    query.WriteString(fmt.Sprintf(` and latest=$%d`, len(args)))
}
```

File: storage/postgres/repository.go (hunk 2 above)
Comment (chakravarthyvp) [reply storage/postgres/repository.go:86]: Usually the SQL driver provides the functions with query parameters (internally validated). If the pg go module does not provide these, then we need to validate it explicitly & raise issue upto the API.
for ex., 
`SELECT id, version, latest from snapshots where namespace=$1 and name=$2` where `name` is technically coming through the API request.

**Comment by kushsharma** [comment]
> Hey, if possible can you separate printcmd changes from download request bug?

I have refactored to fix this bug without introducing `channel` concept.

File: cmd/snapshot.go [hunk 3]
```diff
@@ -93,3 +100,50 @@ func promoteCmd() *cobra.Command {
...
+			if err != nil {
+				return err
+			}
+
+			fds := &descriptorpb.FileDescriptorSet{}
+			if err := proto.Unmarshal(res.GetData(), fds); err != nil {
+				return fmt.Errorf("descriptor set file is not valid. %w", err)
+			}
+
+			fd, err := desc.CreateFileDescriptorFromSet(fds)
```
Comment (harikrishnakanchi) [inline cmd/snapshot.go:131]: `fd` contains only one file descriptor. This is not printing all file details.

**Comment by harikrishnakanchi** [comment]
Couple of options we could add.

- Have a command flag to allow json output
- Allow selector field

**Comment by kushsharma** [comment]
> Couple of options we could add.
> 
> * Have a command flag to allow json output

How are we planning to use `json` output? How will you map `proto` syntax to `json` syntax like comments, rpcs, etc?

> * Allow selector field

What do you mean by selector field, didn't get you? @harikrishnakanchi

File: cmd/snapshot.go (hunk 3 above)
Comment (kushsharma) [reply cmd/snapshot.go]: Should be working now, it also supports printing them to the filesystem via an `output` flag.

**Comment by harikrishnakanchi** [comment]
> How are we planning to use json output? How will you map proto syntax to json syntax like comments, rpcs, etc?

You can use [protojson](https://pkg.go.dev/google.golang.org/protobuf@v1.27.1/encoding/protojson) package to encode to/from json for protobufs.

> What do you mean by selector field, didn't get you?

Something like `--select-by-path=google/protobuf/descriptor.proto` or `--select-by-package-name=google.protobuf`.

File: storage/postgres/repository.go (hunk 2 above)
Comment (harikrishnakanchi) [reply storage/postgres/repository.go:86]: Here the postgres driver that we're using does execute this in prepared statement only. AFAIK prepared statement does avoid SQL injection.

**Comment by kushsharma** [comment]
> You can use [protojson](https://pkg.go.dev/google.golang.org/protobuf@v1.27.1/encoding/protojson) package to encode to/from json for protobufs.

`protojson` is to serialize a protocol buffer message to `json`, it can't convert protobuf DSL to `json`. I don't think it make sense to do it either, how is it useful?

> Something like `--select-by-path=google/protobuf/descriptor.proto` or `--select-by-package-name=google.protobuf`.

Oh okay, like a `filter`. Yeah, we can add filters as well, let me add one for now `--filterby-path=google/protobuf/descriptor.proto`. Does this flag look alright?

**Comment by harikrishnakanchi** [comment]
Not from protobuf DSL. Convert using `descriptorpb.FileDescriptorSet`. You can just pass FileDescriptorSet to  `protojson.Marshal` function directly.

**Comment by kushsharma** [comment]
> Not from protobuf DSL. Convert using `descriptorpb.FileDescriptorSet`. You can just pass FileDescriptorSet to `protojson.Marshal` function directly.

Can you help me understand what will be the use-case for this?

**Comment by harikrishnakanchi** [comment]
JSON provides structured representation, so users can just deal with JSONs to do any processing/debugging without needing to deal with proto libraries. I myself used json representation to debug. that way it's really useful.

**Comment by kushsharma** [comment]
You mean this is useful for debug?
```json
{
  "name": "google/protobuf/timestamp.proto",
  "package": "google.protobuf",
  "messageType": [
    {
      "name": "Timestamp",
      "field": [
        {
          "name": "seconds",
          "number": 1,
          "label": "LABEL_OPTIONAL",
          "type": "TYPE_INT64",
          "jsonName": "seconds"
        },
        {
          "name": "nanos",
          "number": 2,
          "label": "LABEL_OPTIONAL",
          "type": "TYPE_INT32",
          "jsonName": "nanos"
        }
      ]
    }
  ],
  "options": {
    "javaPackage": "com.google.protobuf",
    "javaOuterClassname": "TimestampProto",
    "javaMultipleFiles": true,
    "goPackage": "google.golang.org/protobuf/types/known/timestamppb",
    "ccEnableArenas": true,
    "objcClassPrefix": "GPB",
    "csharpNamespace": "Google.Protobuf.WellKnownTypes"
  },
  "sourceCodeInfo": {
    "location": [
      {
        "span": [
          30,
          0,
          146,
          1
        ]
      },
      {
        "path": [
          12
        ],
        "span": [
          30,
          0,
          18
        ],
        "leadingDetachedComments": [
          " Protocol Buffers - Google's data interchange format\n Copyright 2008 Google Inc.  All rights reserved.\n https://developers.google.com/protocol-buffers/\n\n Redistribution and use in source and binary forms, with or without\n modification, are permitted provided that the following conditions are\n met:\n\n     * Redistributions of source code must retain the above copyright\n notice, this list of conditions and the following disclaimer.\n     * Redistributions in binary form must reproduce the above\n copyright notice, this list of conditions and the following disclaimer\n in the documentation and/or other materials provided with the\n distribution.\n     * Neither the name of Google Inc. nor the names of its\n contributors may be used to endorse or promote products derived from\n this software without specific prior written permission.\n\n THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS\n \"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT\n LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR\n A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT\n OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,\n SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT\n LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,\n DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY\n THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE\n OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n"
        ]
      },
      {
        "path": [
          2
        ],
        "span": [
          32,
          0,
          24
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          34,
          0,
          59
        ]
      },
      {
        "path": [
          8,
          37
        ],
        "span": [
          34,
          0,
          59
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          35,
          0,
          31
        ]
      },
      {
        "path": [
          8,
          31
        ],
        "span": [
          35,
          0,
          31
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          36,
          0,
          73
        ]
      },
      {
        "path": [
          8,
          11
        ],
        "span": [
          36,
          0,
          73
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          37,
          0,
          44
        ]
      },
      {
        "path": [
          8,
          1
        ],
        "span": [
          37,
          0,
          44
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          38,
          0,
          47
        ]
      },
      {
        "path": [
          8,
          8
        ],
        "span": [
          38,
          0,
          47
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          39,
          0,
          34
        ]
      },
      {
        "path": [
          8,
          10
        ],
        "span": [
          39,
          0,
          34
        ]
      },
      {
        "path": [
          8
        ],
        "span": [
          40,
          0,
          33
        ]
      },
      {
        "path": [
          8,
          36
        ],
        "span": [
          40,
          0,
          33
        ]
      },
      {
        "path": [
          4,
          0
        ],
        "span": [
          135,
          0,
          146,
          1
        ],
        "leadingComments": " A Timestamp represents a point in time independent of any time zone or local\n calendar, encoded as a count of seconds and fractions of seconds at\n nanosecond resolution. The count is relative to an epoch at UTC midnight on\n January 1, 1970, in the proleptic Gregorian calendar which extends the\n Gregorian calendar backwards to year one.\n\n All minutes are 60 seconds long. Leap seconds are \"smeared\" so that no leap\n second table is needed for interpretation, using a [24-hour linear\n smear](https://developers.google.com/time/smear).\n\n The range is from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z. By\n restricting to that range, we ensure that we can convert to and from [RFC\n 3339](https://www.ietf.org/rfc/rfc3339.txt) date strings.\n\n # Examples\n\n Example 1: Compute Timestamp from POSIX `time()`.\n\n     Timestamp timestamp;\n     timestamp.set_seconds(time(NULL));\n     timestamp.set_nanos(0);\n\n Example 2: Compute Timestamp from POSIX `gettimeofday()`.\n\n     struct timeval tv;\n     gettimeofday(&tv, NULL);\n\n     Timestamp timestamp;\n     timestamp.set_seconds(tv.tv_sec);\n     timestamp.set_nanos(tv.tv_usec * 1000);\n\n Example 3: Compute Timestamp from Win32 `GetSystemTimeAsFileTime()`.\n\n     FILETIME ft;\n     GetSystemTimeAsFileTime(&ft);\n     UINT64 ticks = (((UINT64)ft.dwHighDateTime) << 32) | ft.dwLowDateTime;\n\n     // A Windows tick is 100 nanoseconds. Windows epoch 1601-01-01T00:00:00Z\n     // is 11644473600 seconds before Unix epoch 1970-01-01T00:00:00Z.\n     Timestamp timestamp;\n     timestamp.set_seconds((INT64) ((ticks / 10000000) - 11644473600LL));\n     timestamp.set_nanos((INT32) ((ticks % 10000000) * 100));\n\n Example 4: Compute Timestamp from Java `System.currentTimeMillis()`.\n\n     long millis = System.currentTimeMillis();\n\n     Timestamp timestamp = Timestamp.newBuilder().setSeconds(millis / 1000)\n         .setNanos((int) ((millis % 1000) * 1000000)).build();\n\n\n Example 5: Compute Timestamp from Java `Instant.now()`.\n\n     Instant now = Instant.now();\n\n     Timestamp timestamp =\n         Timestamp.newBuilder().setSeconds(now.getEpochSecond())\n             .setNanos(now.getNano()).build();\n\n\n Example 6: Compute Timestamp from current time in Python.\n\n     timestamp = Timestamp()\n     timestamp.GetCurrentTime()\n\n # JSON Mapping\n\n In JSON format, the Timestamp type is encoded as a string in the\n [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format. That is, the\n format is \"{year}-{month}-{day}T{hour}:{min}:{sec}[.{frac_sec}]Z\"\n where {year} is always expressed using four digits while {month}, {day},\n {hour}, {min}, and {sec} are zero-padded to two digits each. The fractional\n seconds, which can go up to 9 digits (i.e. up to 1 nanosecond resolution),\n are optional. The \"Z\" suffix indicates the timezone (\"UTC\"); the timezone\n is required. A proto3 JSON serializer should always use UTC (as indicated by\n \"Z\") when printing the Timestamp type and a proto3 JSON parser should be\n able to accept both UTC and other timezones (as indicated by an offset).\n\n For example, \"2017-01-15T01:30:15.01Z\" encodes 15.01 seconds past\n 01:30 UTC on January 15, 2017.\n\n In JavaScript, one can convert a Date object to this format using the\n standard\n [toISOString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)\n method. In Python, a standard `datetime.datetime` object can be converted\n to this format using\n [`strftime`](https://docs.python.org/2/library/time.html#time.strftime) with\n the time format spec '%Y-%m-%dT%H:%M:%S.%fZ'. Likewise, in Java, one can use\n the Joda Time's [`ISODateTimeFormat.dateTime()`](\n http://www.joda.org/joda-time/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTime%2D%2D\n ) to obtain a formatter capable of generating timestamps in this format.\n\n\n"
      },
      {
        "path": [
          4,
          0,
          1
        ],
        "span": [
          135,
          8,
          17
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          0
        ],
        "span": [
          139,
          2,
          20
        ],
        "leadingComments": " Represents seconds of UTC time since Unix epoch\n 1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to\n 9999-12-31T23:59:59Z inclusive.\n"
      },
      {
        "path": [
          4,
          0,
          2,
          0,
          5
        ],
        "span": [
          139,
          2,
          7
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          0,
          1
        ],
        "span": [
          139,
          8,
          15
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          0,
          3
        ],
        "span": [
          139,
          18,
          19
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          1
        ],
        "span": [
          145,
          2,
          18
        ],
        "leadingComments": " Non-negative fractions of a second at nanosecond resolution. Negative\n second values with fractions must still have non-negative nanos values\n that count forward in time. Must be from 0 to 999,999,999\n inclusive.\n"
      },
      {
        "path": [
          4,
          0,
          2,
          1,
          5
        ],
        "span": [
          145,
          2,
          7
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          1,
          1
        ],
        "span": [
          145,
          8,
          13
        ]
      },
      {
        "path": [
          4,
          0,
          2,
          1,
          3
        ],
        "span": [
          145,
          16,
          17
        ]
      }
    ]
  },
  "syntax": "proto3"
}
```

compare to 
```protobuf
// ----
// google/protobuf/timestamp.proto
// ----
// Protocol Buffers - Google's data interchange format
// Copyright 2008 Google Inc.  All rights reserved.
// https://developers.google.com/protocol-buffers/
//
// Redistribution and use in source and binary forms, with or without
// modification, are permitted provided that the following conditions are
// met:
//
//     * Redistributions of source code must retain the above copyright
// notice, this list of conditions and the following disclaimer.
//     * Redistributions in binary form must reproduce the above
// copyright notice, this list of conditions and the following disclaimer
// in the documentation and/or other materials provided with the
// distribution.
//     * Neither the name of Google Inc. nor the names of its
// contributors may be used to endorse or promote products derived from
// this software without specific prior written permission.
//
// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

syntax = "proto3";

package google.protobuf;

option csharp_namespace = "Google.Protobuf.WellKnownTypes";

option cc_enable_arenas = true;

option go_package = "google.golang.org/protobuf/types/known/timestamppb";

option java_package = "com.google.protobuf";

option java_outer_classname = "TimestampProto";

option java_multiple_files = true;

option objc_class_prefix = "GPB";

// A Timestamp represents a point in time independent of any time zone or local
// calendar, encoded as a count of seconds and fractions of seconds at
// nanosecond resolution. The count is relative to an epoch at UTC midnight on
// January 1, 1970, in the proleptic Gregorian calendar which extends the
// Gregorian calendar backwards to year one.
//
// All minutes are 60 seconds long. Leap seconds are "smeared" so that no leap
// second table is needed for interpretation, using a [24-hour linear
// smear](https://developers.google.com/time/smear).
//
// The range is from 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z. By
// restricting to that range, we ensure that we can convert to and from [RFC
// 3339](https://www.ietf.org/rfc/rfc3339.txt) date strings.
//
// # Examples
//
// Example 1: Compute Timestamp from POSIX `time()`.
//
//     Timestamp timestamp;
//     timestamp.set_seconds(time(NULL));
//     timestamp.set_nanos(0);
//
// Example 2: Compute Timestamp from POSIX `gettimeofday()`.
//
//     struct timeval tv;
//     gettimeofday(&tv, NULL);
//
//     Timestamp timestamp;
//     timestamp.set_seconds(tv.tv_sec);
//     timestamp.set_nanos(tv.tv_usec * 1000);
//
// Example 3: Compute Timestamp from Win32 `GetSystemTimeAsFileTime()`.
//
//     FILETIME ft;
//     GetSystemTimeAsFileTime(&ft);
//     UINT64 ticks = (((UINT64)ft.dwHighDateTime) << 32) | ft.dwLowDateTime;
//
//     // A Windows tick is 100 nanoseconds. Windows epoch 1601-01-01T00:00:00Z
//     // is 11644473600 seconds before Unix epoch 1970-01-01T00:00:00Z.
//     Timestamp timestamp;
//     timestamp.set_seconds((INT64) ((ticks / 10000000) - 11644473600LL));
//     timestamp.set_nanos((INT32) ((ticks % 10000000) * 100));
//
// Example 4: Compute Timestamp from Java `System.currentTimeMillis()`.
//
//     long millis = System.currentTimeMillis();
//
//     Timestamp timestamp = Timestamp.newBuilder().setSeconds(millis / 1000)
//         .setNanos((int) ((millis % 1000) * 1000000)).build();
//
//
// Example 5: Compute Timestamp from Java `Instant.now()`.
//
//     Instant now = Instant.now();
//
//     Timestamp timestamp =
//         Timestamp.newBuilder().setSeconds(now.getEpochSecond())
//             .setNanos(now.getNano()).build();
//
//
// Example 6: Compute Timestamp from current time in Python.
//
//     timestamp = Timestamp()
//     timestamp.GetCurrentTime()
//
// # JSON Mapping
//
// In JSON format, the Timestamp type is encoded as a string in the
// [RFC 3339](https://www.ietf.org/rfc/rfc3339.txt) format. That is, the
// format is "{year}-{month}-{day}T{hour}:{min}:{sec}[.{frac_sec}]Z"
// where {year} is always expressed using four digits while {month}, {day},
// {hour}, {min}, and {sec} are zero-padded to two digits each. The fractional
// seconds, which can go up to 9 digits (i.e. up to 1 nanosecond resolution),
// are optional. The "Z" suffix indicates the timezone ("UTC"); the timezone
// is required. A proto3 JSON serializer should always use UTC (as indicated by
// "Z") when printing the Timestamp type and a proto3 JSON parser should be
// able to accept both UTC and other timezones (as indicated by an offset).
//
// For example, "2017-01-15T01:30:15.01Z" encodes 15.01 seconds past
// 01:30 UTC on January 15, 2017.
//
// In JavaScript, one can convert a Date object to this format using the
// standard
// [toISOString()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString)
// method. In Python, a standard `datetime.datetime` object can be converted
// to this format using
// [`strftime`](https://docs.python.org/2/library/time.html#time.strftime) with
// the time format spec '%Y-%m-%dT%H:%M:%S.%fZ'. Likewise, in Java, one can use
// the Joda Time's [`ISODateTimeFormat.dateTime()`](
// http://www.joda.org/joda-time/apidocs/org/joda/time/format/ISODateTimeFormat.html#dateTime%2D%2D
// ) to obtain a formatter capable of generating timestamps in this format.
//
//
message Timestamp {
  // Represents seconds of UTC time since Unix epoch
  // 1970-01-01T00:00:00Z. Must be from 0001-01-01T00:00:00Z to
  // 9999-12-31T23:59:59Z inclusive.
  int64 seconds = 1;

  // Non-negative fractions of a second at nanosecond resolution. Negative
  // second values with fractions must still have non-negative nanos values
  // that count forward in time. Must be from 0 to 999,999,999
  // inclusive.
  int32 nanos = 2;
}
```

Alright adding this under `format` flag. That is, `--format json/proto`, default is `proto`

File: cmd/snapshot.go [hunk 4]
```diff
@@ -93,3 +103,82 @@ func promoteCmd() *cobra.Command {
...
+			protoPrinter := &protoprint.Printer{}
+			for fdName, fd := range fdsMap {
+				switch out {
+				case printStdout:
+					fmt.Printf("\n// %s\n", fdName)
+					fmt.Printf("// ----\n")
+					if err := protoPrinter.PrintProtoFile(fd, os.Stdout); err != nil {
+						return err
+					}
+				case printFilesystem:
```
[the remark this answers is not in the document]
Comment (kushsharma) [reply cmd/snapshot.go]: We can but how do you plan to use a filter then? For example, I use the filter by prefix right now?

[the remark this answers is not in the document]
Comment (kushsharma) [reply cmd/snapshot.go]: updated it

**Pull request merged**